Modules/Core/Mesh/test/itkTriangleMeshCurvatureCalculatorTest.cxx

```diff
@@ -0,0 +1,144 @@
+  using PointType = SphereMeshSourceType::PointType;
+  using VectorType = SphereMeshSourceType::VectorType;
+
+  auto      mySphereMeshSource = SphereMeshSourceType::New();
+  PointType center;
+  center.Fill(0);
+  PointType::ValueType scaleInit_1[3] = { 5, 5, 5 };
+  VectorType           scale = scaleInit_1;
+
+  mySphereMeshSource->SetCenter(center);
+  mySphereMeshSource->SetResolution(1);
+  mySphereMeshSource->SetScale(scale);
+  mySphereMeshSource->Update();
+
+  triangleMesh = mySphereMeshSource->GetOutput();
+
+  curvCalculator->SetTriangleMesh(triangleMesh);
+  curvCalculator->SetCurvatureTypeToGaussian();
+  curvCalculator->Compute();
+
+
+  gaussCurvatureData = curvCalculator->GetGaussCurvatureData();
+
+  // Values obtained using the VTK Gaussian Curvature
+  float v1 = 0.06087285;
+  float v2 = 0.04463759;
+
+  // Test if values are correct for Scale 5 and resolution 1 sphere
+  for (unsigned int k = 0; k < triangleMesh->GetNumberOfPoints(); k++)
+  {
+    if (k < 6)
+    {
+      ITK_TEST_EXPECT_TRUE(itk::Math::AlmostEquals(gaussCurvatureData->GetElement(k), v1));
+    }
+    else
+    {
+      ITK_TEST_EXPECT_TRUE(itk::Math::AlmostEquals(gaussCurvatureData->GetElement(k), v2));
+    }
+  }
+
+
+  // Test the method for different scale. The curvature should decrease for larger sphere.
+  PointType::ValueType scaleInit_2[3] = { 100, 100, 100 };
+  scale = scaleInit_2;
+
+  mySphereMeshSource->SetCenter(center);
+  mySphereMeshSource->SetResolution(1);
+  mySphereMeshSource->SetScale(scale);
+  mySphereMeshSource->Update();
+
+  triangleMesh = mySphereMeshSource->GetOutput();
+  curvCalculator->SetTriangleMesh(triangleMesh);
+  curvCalculator->SetCurvatureTypeToGaussian();
+  curvCalculator->Compute();
+
+  gaussCurvatureData = curvCalculator->GetGaussCurvatureData();
+
+  float v3 = 0.00015218;
+  float v4 = 0.00011159;
+
+  // Test if values are correct for Scale 100 and resolution 1 sphere
+  for (unsigned int k = 0; k < triangleMesh->GetNumberOfPoints(); k++)
+  {
+    if (k < 6)
+    {
+      ITK_TEST_EXPECT_TRUE(itk::Math::AlmostEquals(gaussCurvatureData->GetElement(k), v3));
+    }
+    else
+    {
+      ITK_TEST_EXPECT_TRUE(itk::Math::AlmostEquals(gaussCurvatureData->GetElement(k), v4));
+    }
+  }
+
+  // Test for a vtk mesh using MeshFileReader
+  if (argc > 1)
+  {
+    using ReaderType = itk::MeshFileReader<TriangleMeshType>;
+    ReaderType::Pointer polyDataReader = ReaderType::New();
+    polyDataReader->SetFileName(argv[1]);
+    polyDataReader->Update();
+
+    TriangleMeshType::Pointer inputMesh = polyDataReader->GetOutput();
+    curvCalculator->SetTriangleMesh(inputMesh);
+    curvCalculator->SetCurvatureTypeToGaussian();
+    curvCalculator->Compute();
+
+    gaussCurvatureData = curvCalculator->GetGaussCurvatureData();
+    ITK_TEST_EXPECT_TRUE(gaussCurvatureData->Size() == inputMesh->GetNumberOfPoints())
+  }
+
+  std::cout << "[TEST DONE]" << std::endl;
```

Comment:
Although some contributors see this as unnecessary, I do believe it helps. But please, follow the message proposed by the ITK SW Guide.

Reply:
Still pending.